client/authorization/components/signInForm.js

```diff
@@ -1,65 +1,58 @@
-import React, { Component } from 'react'
-import PropTypes from 'prop-types'
-import {connect} from 'react-redux';
-import * as actions from '../actions'
+import React, {useCallback} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
 import classNames from './scss/signInForm.module.scss'
+import * as selectors from "../selectors";
+import {loginRoutine, hideForm, enterEmail, enterPassword} from "../actions";
+
+const SignInForm = () => {
+    const dispatch = useDispatch();
+    const email = useSelector(selectors.email);
+    const password = useSelector(selectors.password);
+    const handleHideFormClick = () => {
+        dispatch(hideForm());
+    };
 
-class SignInForm extends Component {
-    handleChangedEmail = (event) => {
+    const handleChangedEmail = (event) => {
         const email = event.target.value;
-        this.props.enterEmail({email});
+        dispatch(enterEmail({email}));
     };
 
-    handleChangedPassword = (event) => {
+    const handleChangedPassword = (event) => {
         const password = event.target.value;
-        this.props.enterPassword({password});
-    };
-
-    signIn = () => {
-        const {email, password} = this.props;
-        this.props.login({email, password});
+        dispatch(enterPassword({password}));
     };
 
-    render() {
-        const {email, password, hideForm} = this.props;
-        const {signIn__form, signIn__headline, signIn__input, signIn__button} = classNames;
-        const signIn__button_panel = classNames['signIn__button-panel'];
-
-        return (
-            <div className={signIn__form}>
-                <h3 className={signIn__headline}>LOGIN</h3>
-                <input
-                    type='text'
-                    placeholder="email"
-                    onChange={this.handleChangedEmail}
-                    value={email}
-                    className={signIn__input}/>
-                <input
-                    type='text'
-                    placeholder="password"
-                    onChange={this.handleChangedPassword}
-                    value={password}
-                    className={signIn__input}/>
-
-                <div className={signIn__button_panel}>
-                    <button onClick={this.signIn} className={signIn__button}>Sign in</button>
-                    <button onClick={hideForm} className={signIn__button}>Cancel</button>
-                </div>
+    const handleSignInClick = useCallback(
+        () => dispatch(loginRoutine.trigger({email, password})),
+        [dispatch, email, password]
```

Comment:
loginRoutine.trigger is a dependency also

Reply:
ESLint with "react-hooks/rules-of-hooks" says that loginRoutine.trigger is unnecessary dependency